### PR TITLE
GraphQL-129: Retrieve Customer token

### PR DESCRIPTION
**Pull request by pfantini**
### Description
Create new mutation to retrive customer token by email and password

### Manual testing scenarios

1. Run the following request in GraphiQL:
```
mutation {
    generateCustomerToken(
        email: "customer@example.com"
        password: "password"
    ) {
        token
    }
}
```

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=180) <br/>All committers have signed the CLA.

File: app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+class GenerateCustomerToken implements ResolverInterface
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php]: Pls, provide dockblock for class

File: app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+{
+
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php]: Extra line

File: app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ): Value {
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php:55]: Pls, merge mainline
`Magento\Framework\GraphQl\Query\ResolverInterface` was  changed
In this case we need to return `string` value

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Customer;
+
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+class GenerateCustomerTokenTest extends GraphQlAbstract
+{
+
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php]: Extra line

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by magento-engcom-team** [comment]
@pfantini thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        try {
+            $token = $this->customerTokenService->createCustomerAccessToken($args['email'], $args['password']);
+            $result = function () use ($token) {
+                return !empty($token) ? $token : '';
```
Comment (TomashKhamlai) [inline app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php]: Please change to
return !empty($token) ? ['token' => $token] : '';

or `Cannot return null for non-nullable field GenerateCustomerTokenOutput.token.` error will appear

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        $userName = 'customer@example.com';
+        $password = 'bad-password';
+
+        $mutation
+            = <<<MUTATION
+mutation {
+	generateCustomerToken(
+        email: "{$userName}"
+        password: "{$password}"
+    )
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php]: Please add `token` to subselection.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        $userName = 'customer@example.com';
+        $password = 'password';
+
+        $mutation
+            = <<<MUTATION
+mutation {
+	generateCustomerToken(
+        email: "{$userName}"
+        password: "{$password}"
+    )
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php]: Please add `token` to subselection.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+}
+MUTATION;
+
+        $response = $this->graphQlQuery($mutation);
+        $this->assertArrayHasKey('generateCustomerToken', $response);
+        $this->assertInternalType('string', $response['generateCustomerToken']);
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/GenerateCustomerTokenTest.php]: Please change 'string' to 'array'

File: app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php (hunk 3 above)
Comment (pfantini) [reply app/code/Magento/CustomerGraphQl/Model/Resolver/Customer/Account/GenerateCustomerToken.php:55]: If I change returned value to string or array the test `Magento\GraphQl\IntrospectionQueryTest::testIntrospectionQueryWithFieldArgs` fails

**Pull request merged**